src/main/scala/is/hail/methods/LinearMixedRegression.scala

```diff
@@ -23,19 +26,19 @@ object LinearMixedRegression {
     ("pval", TDouble))
 
   def apply(
-    assocVds: VariantDataset,
-    kinshipMatrix: KinshipMatrix,
-    yExpr: String,
-    covExpr: Array[String],
-    useML: Boolean,
-    rootGA: String,
-    rootVA: String,
-    runAssoc: Boolean,
-    optDelta: Option[Double],
-    sparsityThreshold: Double,
-    useDosages: Boolean,
-    optNEigs: Option[Int],
-    optDroppedVarianceFraction: Option[Double]): VariantDataset = {
+             assocVds: VariantDataset,
+             relationMatrix: LMMMatrix,
```

Comment:
How about similarityMatrix, since it can be either sample similarity or variant similarity? Relation feels far more connected to sample than variant. And I think the class should be SimilarityMatrix, as this isn't specific to LMM. You should check with Dan about the class organization too as he's thinking about making a general class for symmetric matrices like these.